### PR TITLE
platformio 3.4.0

### PR DESCRIPTION
**Pull request by ivankravets**
# Release Notes

* [PIO Unified Debugger](http://docs.platformio.org/page/plus/debugging.html)

  - "1-click" solution, zero configuration
  - Support for 100+ embedded boards
  - Multiple architectures and development platforms
  - Windows, MacOS, Linux (+ARMv6-8)
  - Built-in into [PlatformIO IDE for Atom](http://docs.platformio.org/page/ide/atom.html) and [PlatformIO IDE for VScode](http://docs.platformio.org/page/ide/vscode.html)
  - Integration with [Eclipse](http://docs.platformio.org/page/ide/eclipse.html) and [Sublime Text](http://docs.platformio.org/page/ide/sublimetext.html)

* Filter [PIO Unit Testing](http://docs.platformio.org/page/plus/unit-testing.html) tests using a new ``test_filter`` option in [Project Configuration File "platformio.ini"](http://docs.platformio.org/page/projectconf.html) or [platformio test --filter](http://docs.platformio.org/page/userguide/cmd_test.html#cmdoption-platformio-test-f) command ([issue #934](https://github.com/platformio/platformio-core/issues/934))
* Custom ``test_transport`` for [PIO Unit Testing](http://docs.platformio.org/page/plus/unit-testing.html) Engine
* Configure Serial Port Monitor in [Project Configuration File "platformio.ini"](http://docs.platformio.org/page/projectconf.html) ([issue #787](https://github.com/platformio/platformio-core/issues/787))
* New [monitor](http://docs.platformio.org/page/userguide/cmd_run.html#cmdoption-platformio-run-t) target which allows to launch Serial Monitor automatically after successful "build" or "upload" operations ([issue #788](https://github.com/platformio/platformio-core/issues/788))
* Project generator for [VIM](http://docs.platformio.org/page/ide/vim.html)
* Multi-line support for the different options in [Project Configuration File "platformio.ini"](http://docs.platformio.org/page/projectconf.html), such as: ``build_flags``, ``build_unflags``, etc. ([issue #889](https://github.com/platformio/platformio-core/issues/889))
* Handle dynamic ``SRC_FILTER`` environment variable from [library.json extra script](http://docs.platformio.org/page/librarymanager/config.html#extrascript)
* Notify about multiple installations of PIO Core ([issue #961](https://github.com/platformio/platformio-core/issues/961))
* Improved auto-detecting of mbed-enabled media disks
* Automatically update Git-submodules for development platforms and libraries that were installed from repository
* Add support for ``.*cc`` extension ([issue #939](https://github.com/platformio/platformio-core/issues/939))
* Handle ``env_default`` in [Project Configuration File "platformio.ini"](http://docs.platformio.org/page/projectconf.html) when re-initializing a project ([issue #950](https://github.com/platformio/platformio-core/issues/950))
* Use root directory for PIO Home when path contains non-ascii characters ([issue #951](https://github.com/platformio/platformio-core/issues/951), [issue #952](https://github.com/platformio/platformio-core/issues/952))
* Don't warn about known ``boards_dir`` option ([pull #949](https://github.com/platformio/platformio-core/pull/949))
* Escape non-valid file name characters when installing a new package (library) ([issue #985](https://github.com/platformio/platformio-core/issues/985))
* Fixed infinite dependency installing when repository consists of multiple libraries ([issue #935](https://github.com/platformio/platformio-core/issues/935))
* Fixed linter error "unity.h does not exist" for Unit Testing ([issue #947](https://github.com/platformio/platformio-core/issues/947))
* Fixed issue when [Library Dependency Finder (LDF)](http://docs.platformio.org/page/librarymanager/ldf.html) does not handle custom ``src_dir`` ([issue #942](https://github.com/platformio/platformio-core/issues/942))
* Fixed cloning a package (library) from a private Git repository with custom user name and SSH port ([issue #925](https://github.com/platformio/platformio-core/issues/925))

### CODE REVIEW

**Comment by ilovezfs** [comment]
It looks like you're missing some dependencies. I recommend that you use https://github.com/tdsmith/homebrew-pypi-poet/ to fix that.

**Comment by ivankravets** [comment]
Thank you very much! I updated formula with poet's output.

**Comment by ilovezfs** [comment]
@ivankravets you're welcome. Note that the bottle block should not be altered. The test-bot handles the commits that update the bottle block automatically.

**Comment by ivankravets** [comment]
I reverted back "bottle" block

File: Formula/platformio.rb [hunk 1]
```diff
@@ -13,31 +13,41 @@ class Platformio < Formula
     sha256 "ffbfedc16ce2eccbb65ca74b48cf6c94cf24097b0fd88d70f7bc323a1435fd7f" => :yosemite
   end
 
-  depends_on :python if MacOS.version <= :snow_leopard
```
Comment (ilovezfs) [inline Formula/platformio.rb]: This should stay as it was.

File: Formula/platformio.rb [hunk 2]
```diff
@@ -13,31 +13,41 @@ class Platformio < Formula
...
-
+  depends_on :python
+
```
Comment (ilovezfs) [inline Formula/platformio.rb]: white space issue here

File: Formula/platformio.rb [hunk 3]
```diff
@@ -46,26 +56,42 @@ class Platformio < Formula
...
+  end  
+
   def install
     virtualenv_install_with_resources
   end
 
   test do
     system bin/"platformio"
     system bin/"pio"
+    system bin/"piodebuggdb"
```
Comment (ilovezfs) [inline Formula/platformio.rb]: The exit code is 1 not 0, so you'll need to use `shell_output("#{bin}/piodebuggdb", 1)`, and `assert_match` against some part of the output if you want to test this command.

**Comment by ivankravets** [comment]
Sorry, why we should use this `poet` when it generates incorrect formula?

**Comment by ilovezfs** [comment]
>Sorry, why we should use this poet when it generates incorrect formula?

Once a formula already exists, usually `poet -r` is sufficient.

File: Formula/platformio.rb [hunk 4]
```diff
@@ -46,26 +56,42 @@ class Platformio < Formula
...
+  end  
+
   def install
     virtualenv_install_with_resources
   end
 
   test do
-    system bin/"platformio"
-    system bin/"pio"
+    shell_output("#{bin}/platformio", 1)
```
Comment (ilovezfs) [inline Formula/platformio.rb]: Sorry I meant only piodebuggdb. The other two still have exit code 0.

File: Formula/platformio.rb [hunk 5]
```diff
@@ -46,26 +56,42 @@ class Platformio < Formula
...
+    shell_output("#{bin}/pio", 1)
+    shell_output("#{bin}/piodebuggdb", 1)
```
Comment (ilovezfs) [inline Formula/platformio.rb]: should use assert_match here as well

File: Formula/platformio.rb (hunk 4 above)
Comment (ivankravets) [inline Formula/platformio.rb]: Strange machinery...

File: Formula/platformio.rb (hunk 3 above)
Comment (ivankravets) [inline Formula/platformio.rb]: How we can call with `piodebuggdb -- help`?

**Comment by ilovezfs** [comment]
>How we can call with piodebuggdb -- help?

Yes that will work.

```
system bin/"piodebuggdb", "--help"
```

**Comment by ivankravets** [comment]
Thanks! Now it looks good.

File: Formula/platformio.rb [hunk 6]
```diff
@@ -46,26 +56,42 @@ class Platformio < Formula
...
 
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz"
+    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/96/d9/40e4e515d3e17ed0adbbde1078e8518f8c4e3628496b56eb8f026a02b9e4/urllib3-1.21.1.tar.gz"
+    sha256 "b14486978518ca0901a76ba973d7821047409d7f726f22156b24e83fd71382a5"
+  end
```
Comment (ilovezfs) [inline Formula/platformio.rb]: There's trailing whitespace here

**Comment by ilovezfs** [comment]
:green_apple:

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for the update @ivankravets! :sparkle:

**Comment by ivankravets** [comment]
Thank you too!

**Pull request closed without merging**